fluentd/configs.d/filter-pre-mux-client.conf

```diff
@@ -5,6 +5,8 @@
   shared_key "#{File.open('/etc/fluent/muxkeys/shared_key') do |f| f.readline end.rstrip}"
   secure yes
   ca_cert_path /etc/fluent/muxkeys/ca
+  buffer_queue_limit "#{ENV['BUFFER_QUEUE_LIMIT']}"
```

Comment:
@portante
> If a buffer_queue_limit is reached, then an exception is raised, and the json-file reader or journald reader should stop reading more data until it can place data into the buffer queue. Let's assert that that is the behavior we want going forward.
Sorry for my ignorance...  This feature is going to be in the newer version of fluentd?

> If we are going to stick with the memory buffer, which is arguably a bad idea, ...
I googled around and found configuring with the file buffer does not show much performance degradation, which is promising.  And that's the only a concern, isn't it?

Do we want to make the buffering type configurable?  Or permanently switching to the file buffer is good enough?

> With a low flush_interval, it is possible that chunks are queued to be flushed before they are filled.
I think that's the status we are hoping.  To realize it, I'd think we would like to guarantee that the ElasticSearch has enough CPU and memory.  Another thought is when we have 2 ElasticSearch pods (one for project; another for .operations), setting 2 to num_threads and see how speeds up the flushing.  (But fluentd would require more resources...)
http://docs.fluentd.org/v0.12/articles/out_file#numthreads